packages/flame/lib/src/components/component.dart

```diff
@@ -131,34 +131,28 @@ class Component with Loadable {
     children.forEach((c) => c.update(dt));
   }
 
-  @mustCallSuper
-  void render(Canvas canvas) {
-    preRender(canvas);
-  }
+  void render(Canvas canvas) {}
 
-  @mustCallSuper
   void renderTree(Canvas canvas) {
+    preRender(canvas);
     render(canvas);
-    postRender(canvas);
-    children.forEach((c) {
-      canvas.save();
-      c.renderTree(canvas);
-      canvas.restore();
-    });
+    children.forEach((c) => c.renderTree(canvas));
 
     // Any debug rendering should be rendered on top of everything
     if (debugMode) {
       renderDebugMode(canvas);
     }
+    postRender(canvas);
```

Comment:
I think this should be before `renderDebugMode`, so that the debug rendering always is on top.

Reply:
Ahh, the `restore` runs in there now.